### PR TITLE
[ISSUE-851][REFACTOR] Migrate quota configruation to Celeborn Configuration System

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
@pan3793 @waitinfuture

**Comment by waitinfuture** [comment]
> 

In future we might have quota on other resource types such as memory and network. I think we'd better use quota.storage.xxx, quota.memory.xxx

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -1625,6 +1635,46 @@ object CelebornConf extends Logging {
       .checkValue(p => p >= 1024 && p < 65535, "invalid port")
       .createWithDefault(9096)
 
+  val SHUFFLE_CHECK_QUOTA_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.shuffle.checkQuota.enabled")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1639]: celeborn.quota.enabled

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -1625,6 +1635,46 @@ object CelebornConf extends Logging {
...
+      .categories("quota")
+      .doc("When true, before registering shuffle, LifecycleManager should check " +
+        "if current user have enough quota space, if cluster don't have enough " +
+        "quota space for current user, fallback to Spark's default shuffle")
+      .version("0.2.0")
+      .booleanConf
+      .createWithDefault(true)
+
+  val SHUFFLE_IDENTITY_PROVIDER: ConfigEntry[String] =
+    buildConf("celeborn.shuffle.identity.provider")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1650]: celeborn.quota.identity.provider

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 3]
```diff
@@ -1625,6 +1635,46 @@ object CelebornConf extends Logging {
...
+      .categories("quota")
+      .doc(s"IdentityProvider class name. Default class is " +
+        s"`${classOf[DefaultIdentityProvider].getName}`, return `${classOf[UserIdentifier].getName}` " +
+        s"with default tenant id and username from `${classOf[UserGroupInformation].getName}`. ")
+      .version("0.2.0")
+      .stringConf
+      .createWithDefault(classOf[DefaultIdentityProvider].getName)
+
+  val SHUFFLE_QUOTA_MANAGER: ConfigEntry[String] =
+    buildConf("celeborn.shuffle.quota.manager")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1661]: celeborn.quota.manager

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 4]
```diff
@@ -1625,6 +1635,46 @@ object CelebornConf extends Logging {
...
+      .withAlternative("rss.quota.manager")
+      .categories("quota")
+      .doc(s"QuotaManger class name. Default class is `${classOf[DefaultQuotaManager].getName}`.")
+      .version("0.2.0")
+      .stringConf
+      .createWithDefault(classOf[DefaultQuotaManager].getName)
+
+  val SHUFFLE_QUOTA_CONFIGURATION_PATH: OptionalConfigEntry[String] =
+    buildConf("celeborn.shuffle.quota.configuration.path")
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1670]: celeborn.quota.configuration.path

**Comment by AngersZhuuuu** [comment]
All done, how about current

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 5]
```diff
@@ -499,6 +501,14 @@ class CelebornConf(loadDefaults: Boolean) extends Cloneable with Logging with Se
   def workerPrometheusMetricHost: String = get(WORKER_PROMETHEUS_HOST)
   def workerPrometheusMetricPort: Int = get(WORKER_PROMETHEUS_PORT)
 
+  // //////////////////////////////////////////////////////
+  //                      Quota                         //
+  // //////////////////////////////////////////////////////
+  def checkQuotaEnabled: Boolean = get(CHECK_QUOTA_ENABLED)
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:507]: quotaEnabled

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 6]
```diff
@@ -499,6 +501,14 @@ class CelebornConf(loadDefaults: Boolean) extends Cloneable with Logging with Se
...
+  def identityProviderClass: String = get(IDENTITY_PROVIDER)
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:508]: quotaIdentityProviderClass

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 7]
```diff
@@ -1625,6 +1635,46 @@ object CelebornConf extends Logging {
       .checkValue(p => p >= 1024 && p < 65535, "invalid port")
       .createWithDefault(9096)
 
+  val CHECK_QUOTA_ENABLED: ConfigEntry[Boolean] =
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1638]: QUOTA_ENABLED

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 8]
```diff
@@ -1625,6 +1635,46 @@ object CelebornConf extends Logging {
...
+      .withAlternative("rss.cluster.checkQuota.enabled")
+      .categories("quota")
+      .doc("When true, before registering shuffle, LifecycleManager should check " +
+        "if current user have enough quota space, if cluster don't have enough " +
+        "quota space for current user, fallback to Spark's default shuffle")
+      .version("0.2.0")
+      .booleanConf
+      .createWithDefault(true)
+
+  val IDENTITY_PROVIDER: ConfigEntry[String] =
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1649]: QUOTA_IDENTITY_PROVIDER

File: common/src/test/scala/org/apache/celeborn/ConfigurationSuite.scala [hunk 9]
```diff
@@ -74,6 +74,10 @@ class ConfigurationSuite extends AnyFunSuite {
     generateConfigurationMarkdown("metrics")
   }
 
+  test("docs - configuration/quota.md") {
```
Comment (pan3793) [inline common/src/test/scala/org/apache/celeborn/ConfigurationSuite.scala:77]: nit: move before "docs - configuration/metrics.md"

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 10]
```diff
@@ -1625,6 +1635,46 @@ object CelebornConf extends Logging {
       .checkValue(p => p >= 1024 && p < 65535, "invalid port")
       .createWithDefault(9096)
 
+  val QUOTA_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.checkQuota.enabled")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1639]:
```suggestion
    buildConf("celeborn.quota.enabled")
```

**Review by pan3793 — APPROVED** [review]

**Pull request merged**